src/core.ts

```diff
@@ -97,29 +98,65 @@ export interface Options {
   killSignal?:    NodeJS.Signals
   halt?:          boolean
 }
-// prettier-ignore
-export const defaults: Options = {
-  [CWD]:          process.cwd(),
-  [SYNC]:         false,
-  verbose:        false,
-  env:            process.env,
-  sync:           false,
-  shell:          true,
-  stdio:          'pipe',
-  nothrow:        false,
-  quiet:          false,
-  prefix:         '',
-  postfix:        '',
-  detached:       false,
-  preferLocal:    false,
-  spawn,
-  spawnSync,
-  log,
-  kill,
-  killSignal:     SIGTERM,
-  timeoutSignal:  SIGTERM,
+
+export function getZxDefaults(
+  defs: Options,
+  prefix: string = 'ZX_',
+  env = process.env
+) {
+  const types: Record<PropertyKey, Array<'string' | 'boolean'>> = {
+    cwd: ['string'],
+    shell: ['string', 'boolean'],
+    halt: ['boolean'],
+    preferLocal: ['string', 'boolean'],
+    input: ['string'],
+  }
+
+  const o = Object.entries(env).reduce<Record<string, string | boolean>>(
+    (m, [k, v]) => {
+      if (k.startsWith(prefix) && v) {
+        const _k = snakeToCamel(k.slice(prefix.length))
+        const _v = { true: true, false: false }[v.toLowerCase()] ?? v
+        if (
+          [typeof defs[_k as keyof Options], ...(types[_k] ?? [])].includes(
```

Comment:
`{...undefined}`
undefined is spreadable, afair, isnt it?

Reply:
I thought the same, but output from my Browser console

```
[...undefined]
VM153:1 Uncaught TypeError: undefined is not iterable
    at <anonymous>:1:5
 ```

Reply:
but in your case it works)

```
> {...undefined}
< {}
```